### PR TITLE
Cb tie game logic

### PR DESCRIPTION
**Pull request by Cristian-Baeza**

### CODE REVIEW

File: lib/main_app.rb [hunk 1]
```diff
@@ -9,19 +9,23 @@ def run
     @game = GameLogic.new
     @console_output = ConsoleOutput.new
 
-    until @game.is_there_winner? do
+    until @game.board_full? || @game.is_there_winner? do
```
Comment (td-extreme) [inline lib/main_app.rb:12]: Can we change this to ?
```suggestion
    until @game.is_over? do
```
and then add that method to GameLogic

File: lib/main_app.rb [hunk 2]
```diff
@@ -9,19 +9,23 @@ def run
...
-      @game.take_turn(user_choice) 
+      @game.take_turn(user_choice)
+    end
 
+    if @game.is_there_winner?
+      puts(@console_output.show_board(@game)) 
+      puts("GAME OVER: #{@game.is_there_winner?} WINS!!")
+    else
+      puts(@console_output.show_board(@game))
+      puts(@console_output.tie_message)
```
Comment (td-extreme) [inline lib/main_app.rb:27]:
```suggestion
    puts(@console_output.show_board(@game)) 
    if @game.is_there_winner?
      puts("GAME OVER: #{@game.is_there_winner?} WINS!!")
    else
      puts(@console_output.tie_message)
```
You output the gameboard in both the if and else block. Let's move that above to remove duplication.

**Review by td-extreme — CHANGES_REQUESTED** [review]
Looks good. I just have some cosmetic concerns.

File: lib/main_app.rb (hunk 1 above)
Comment (Cristian-Baeza) [reply lib/main_app.rb:12]: 👍🏽

File: lib/console_output.rb [hunk 3]
```diff
@@ -11,6 +12,10 @@ def space_taken_message
       SPACE_TAKEN
     end
 
+    def tie_message
```
[the remark this answers is not in the document]
Comment (Cristian-Baeza) [reply lib/console_output.rb:15]: 👍🏽

**Pull request merged**